packages/client/src/generation/TSClient/Count.ts

```diff
@@ -219,7 +219,7 @@ ${indent(
     .map((f) => {
       const fieldTypeName = (f.outputType.type as DMMF.OutputType).name
       return `
-${f.name}<T extends ${getFieldArgName(f)} = {}>(args?: Subset<T, ${getFieldArgName(f)}>): ${getSelectReturnType({
+${f.name}<T extends ${getFieldArgName(f)}>(args?: Subset<T, ${getFieldArgName(f)}>): ${getSelectReturnType({
```

Comment:
Since `args` is an optional parameter, will TypeScript be able to infer anything when the argument is not passed to the function?

Reply:
Another option might have been using `${getFieldArgName(f)}` instead of `{}` as the default value to get all properties by default, but that would probably break the `CheckSelect` check in the return type generated by `getSelectReturnType`.

Reply:
> Since args is an optional parameter, will TypeScript be able to infer anything when the argument is not passed to the function?

Actually it will infer it to the type in the `extends` clause (i.e., `${getFieldArgName(f)}` in this case), yeah.